Zend/zend_operators.h

```diff
@@ -267,13 +280,16 @@ ZEND_API void ZEND_FASTCALL convert_to_boolean(zval *op);
 ZEND_API void ZEND_FASTCALL convert_to_array(zval *op);
 ZEND_API void ZEND_FASTCALL convert_to_object(zval *op);
 
-ZEND_API zend_long    ZEND_FASTCALL zval_get_long_func(zval *op);
+ZEND_API zend_long    ZEND_FASTCALL zval_get_long_func(zval *op, bool is_lax);
 ZEND_API double       ZEND_FASTCALL zval_get_double_func(zval *op);
 ZEND_API zend_string* ZEND_FASTCALL zval_get_string_func(zval *op);
 ZEND_API zend_string* ZEND_FASTCALL zval_try_get_string_func(zval *op);
 
 static zend_always_inline zend_long zval_get_long(zval *op) {
-	return EXPECTED(Z_TYPE_P(op) == IS_LONG) ? Z_LVAL_P(op) : zval_get_long_func(op);
+	return EXPECTED(Z_TYPE_P(op) == IS_LONG) ? Z_LVAL_P(op) : zval_get_long_func(op, 0);
```

Comment:
This looks like a much larger change than what is described in the RFC. Doesn't this make all manual type conversions in extensions "strict"? We have always treated those the same way as explicit casts.

Reply:
Oh I didn't know it was treated like a cast, will change this then again

Reply:
FWIW it might make sense to change that, but this probably isn't the place to start. Throwing a warning about float truncation while still happily and silently interpreting `[]` and `"foobar"` as `0` would be rather incongruent.

Reply:
Yeah, but this would be an RFC to change the behaviour of userland casts I imagine?
I was considering anyway to make an (int) cast throw a notice/warning when casting a float larger than INT_MAX to align string float and normal float behaviour, but that's a rather separate change again.